starknet_py/net/contract/sent_transaction.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+import dataclasses
+import warnings
+from dataclasses import dataclass
+from typing import Optional, TypeVar
+
+from starknet_py.net.client import Client
+from starknet_py.utils.sync import add_sync_methods
+
+
+@add_sync_methods
+@dataclass(frozen=True)
+class SentTransaction:
+    """
+    Dataclass exposing the interface of transaction related to a performed action.
+    """
+
+    hash: int
+    """Hash of the transaction."""
+
+    _client: Client
+    status: Optional[str] = None
+    """Status of the transaction."""
+
+    block_number: Optional[int] = None
+    """Number of the block in which transaction was included."""
+
+    async def wait_for_acceptance(
+        self: TypeSentTransaction,
+        wait_for_accept: Optional[bool] = None,
+        check_interval: float = 2,
+        retries: int = 500,
+    ) -> TypeSentTransaction:
+        """
+        Waits for transaction to be accepted on chain till ``ACCEPTED`` status.
+        Returns a new SentTransaction instance, **does not mutate original instance**.
+        """
+        if wait_for_accept is not None:
+            warnings.warn(
+                "Parameter `wait_for_accept` has been deprecated - since Starknet 0.12.0, transactions in a PENDING"
+                " block have status ACCEPTED_ON_L2."
+            )
```

Comment:
Current starknet.py version is not compatible with 0.12.0 anymore, so we can probably remove this, right?

Reply:
Will be done in https://github.com/software-mansion/starknet.py/issues/1268, as there is the same parameter to remove in the `Client.wait_for_tx()`.